### DISCUSSION TITLE
this is of type any in actions using @pinia/nuxt

### DISCUSSION THREAD

**Discussion by P4sca1** [vuejs/pinia#1451] [Help and Questions] [answered]
### Reproduction

https://github.com/P4sca1/nuxt-pinia-action-type-this

### Steps to reproduce the bug

1. Run `yarn` to install dependencies
2. Run `yarn dev` to generate the `.nuxt` folder
3. View file `stores/counter.ts`
4. Hower over `this` in line 14

### Expected behavior

`this` should be properly typed.

### Actual behavior

`this` is of type `any`.

### Additional information

`this` is properly typed in actions using plain vue.js with pinia (without nuxt).

**Comment by posva** [comment, marked as answer]
You need to add `strict: true` to your tsconfig.json for pinia to work. Maybe worth reporting in Nuxt as `strict: true` is a good practice.

**Reply by P4sca1** [reply]
I can confirm that updating my `tsconfig.json` file resolved the issue. Thank you for your quick answer.
```json
{
	// https://v3.nuxtjs.org/concepts/typescript
	"extends": "./.nuxt/tsconfig.json",
	"compilerOptions": {
		"strict": true
	}
}
```
The reason it worked in plain Vue.js + pinia is that `npm init vue@latest` defaults to `strict: true`: https://github.com/vuejs/tsconfig/blob/26fc528db347dbbbbaa2ea15d3c2d95c197887b7/tsconfig.json#L15.
